ts/features/idpay/onboarding/utility/hooks.ts

```diff
@@ -0,0 +1,66 @@
+/* eslint-disable functional/immutable-data */
+import { useSelector } from "@xstate/react";
+import { useEffect, useRef, useState } from "react";
+import {
+  NativeScrollEvent,
+  ScrollView,
+  useWindowDimensions
+} from "react-native";
+import { useOnboardingMachineService } from "../xstate/provider";
+import {
+  initiativeDescriptionSelector,
+  isLoadingSelector
+} from "../xstate/selectors";
+
+export const useInitiativeDetailsScrolling = () => {
+  const machine = useOnboardingMachineService();
```

Comment:
Scrolling is just a UI event i don't think it's a good idea to bind it with the machine.

Reply:
right now the machine is used to determine when loading states and the description's loading are over.

Do you think it would be more idiomatic to move the screen's machine-related logic in the hook and only expose to the component the bare minimum to handle the scrolling, moving said logic there (or vice versa)?

surely there would be a better division of concerns between the two, but I believe it would bring some unnatural patterns